### DISCUSSION TITLE
How can I use custom paths instead of @site?

### DISCUSSION THREAD

**Discussion by soosap** [facebook/docusaurus#9551] [General]
We are using docusaurus inside a monorepo with all the projects following this convention inside `tsconfig`:

```json
{
  "compilerOptions": {
    "baseUrl": ".",
    "paths": {
      "@/*": ["./src/*"]
    }
  }
}
```

It is kind of confusing that in all our projects we can use this syntax:

```tsx
import HomepageFeatures from '@/components/HomepageFeatures'
```

Except for Docusaurus we have to use what is being dictated:
```tsx
import HomepageFeatures from '@site/src/components/HomepageFeatures'
```

Is there a way we could use our own "path" definitions. Right now the `tsconfig.json` at the root of the project is completely ignored during `docusaurus build` and `docusaurus start`.

**Comment by Josh-Cena** [comment]
Docusaurus doesn't read tsconfig. You need to inject custom Webpack config through [a custom plugin](https://docusaurus.io/docs/api/docusaurus-config#plugins) that uses [`configureWebpack`](https://docusaurus.io/docs/api/plugin-methods/lifecycle-apis#configureWebpack).
